### ISSUE TITLE
Caja crashes when install python-caja but python2-caja init fail

### ISSUE THREAD

**Issue by eaglexmw** [mate-desktop/caja#451]
Caja crashes when install python-caja but python2-caja init fail

bug code in libcaja-private/caja-module.c

static void
add_module_objects (CajaModule *module)
{
    GObject *object;
    const GType *types;
    int num_types;
    int i;

    module->list_types (&types, &num_types);

    for (i = 0; i < num_types; i++)
    {
        if (types[i] == 0)   /* Work around broken extensions */
        {
            break;
        }
        object = caja_module_add_type (types[i]);
    }
    gchar *filename = g_path_get_basename (module->path);
    caja_extension_register (filename, object);
}

list_types function will return nothing, but call caja_extension_register ...

i repair it by this patch, it work fine.

```diff
diff -Nur old/libcaja-private/caja-module.c new/libcaja-private/caja-module.c
--- old/libcaja-private/caja-module.c	2014-12-29 10:27:41.000000000 +0000
+++ new/libcaja-private/caja-module.c	2015-08-09 10:38:48.217917872 +0000
@@ -154,10 +154,12 @@
 {
     GObject *object;
     const GType *types;
-    int num_types;
+    int num_types = 0;
     int i;
 
     module->list_types (&types, &num_types);
+    if (num_types == 0)
+        return;
 
     for (i = 0; i < num_types; i++)
     {
```

**Comment by monsta** [comment]
Unfortunately, this leads to skipping of ```python-caja``` when extensions are being loaded.

**Comment by monsta** [comment]
Ok, this is now properly fixed in Caja 1.12.6.
